### PR TITLE
Tidy variables in clusterctl templating 

### PR DESCRIPTION
**Pull request by macaptain**
Refactoring so that variables which generate the machine templates for tests are not needlessly duplicated by Ansible, environment variables and clusterctl.

Some advantages of doing this:
* Entire bash scripts not embedded in yaml inside environment variables
* No need to escape $ and " signs inside scripts and syntax highlighting works for the scripts in vim
* Remove unnecessary duplication of environment variables vs. Ansible variables
* Environment used to generate the clusterctl config is now captured entirely in ~/.cluster-api/
* Make clear which config is used only in the upgrade test

### CODE REVIEW

File: vm-setup/roles/v1aX_integration_test/templates/cluster-template-controlplane.yaml [hunk 1]
```diff
@@ -43,7 +43,9 @@ spec:
           cgroup-driver: "systemd"
           container-runtime-endpoint: "unix:///var/run/crio/crio.sock"
           runtime-request-timeout: "5m"
-${ CTLPLANE_KUBEADM_EXTRA_CONFIG }
+{% filter indent(width=4) %}
+{% include "cluster-template-controlplane-kubeadm-config-%s.yaml" % OS_RELEASE %}
+{% endfilter %}
```
Comment (lentzi90) [inline vm-setup/roles/v1aX_integration_test/templates/cluster-template-controlplane.yaml:48]: TIL! Cool :sunglasses:

File: vm-setup/roles/v1aX_integration_test/templates/clusterctl_env_template_centos.rc [hunk 2]
```diff
@@ -1,226 +0,0 @@
...
-export PROVISIONING_CIDR='{{ PROVISIONING_CIDR }}'
-export BAREMETALV4_POOL_RANGE_START='{{ BAREMETALV4_POOL_RANGE_START }}'
-export BAREMETALV4_POOL_RANGE_END='{{ BAREMETALV4_POOL_RANGE_END }}'
-export EXTERNAL_SUBNET_V4_PREFIX='{{ EXTERNAL_SUBNET_V4_PREFIX }}'
-export EXTERNAL_SUBNET_V4_HOST='{{ EXTERNAL_SUBNET_V4_HOST }}'
-export BAREMETALV6_POOL_RANGE_START='{{ BAREMETALV6_POOL_RANGE_START }}'
-export BAREMETALV6_POOL_RANGE_END='{{ BAREMETALV6_POOL_RANGE_END }}'
-export EXTERNAL_SUBNET_V6_PREFIX='{{ EXTERNAL_SUBNET_V6_PREFIX }}'
-export EXTERNAL_SUBNET_V6_HOST='{{ EXTERNAL_SUBNET_V6_HOST }}'
-export CTLPLANE_KUBEADM_EXTRA_CONFIG="
```
Comment (lentzi90) [inline vm-setup/roles/v1aX_integration_test/templates/clusterctl_env_template_centos.rc:26]: Oh what a relief to get this out of an environment variable and into its own file! :relieved:

File: vm-setup/roles/v1aX_integration_test/templates/cluster-template-controlplane-kubeadm-config-centos.yaml [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  - systemctl restart NetworkManager.service
+  - ifup eth0
+  - systemctl enable --now crio keepalived kubelet
+  - systemctl link /lib/systemd/system/monitor.keepalived.service
+  - systemctl enable monitor.keepalived.service
+  - systemctl start monitor.keepalived.service
+postKubeadmCommands:
+  - mkdir -p /home/{{ IMAGE_USERNAME }}/.kube
+  - cp /etc/kubernetes/admin.conf /home/{{ IMAGE_USERNAME }}/.kube/config
+  - chown {{ IMAGE_USERNAME }}:{{ IMAGE_USERNAME }} /home/{{ IMAGE_USERNAME }}/.kube/config
```
Comment (lentzi90) [inline vm-setup/roles/v1aX_integration_test/templates/cluster-template-controlplane-kubeadm-config-centos.yaml:16]: I realize there was no change here, but I'm curious why this is needed. Don't we get the kubeconfig from a Secret in the management cluster?

**Review by lentzi90 — APPROVED** [review]
Great! Looking forward to getting this merged! :smile:

File: vm-setup/roles/v1aX_integration_test/templates/cluster-template-controlplane-kubeadm-config-centos.yaml (hunk 3 above)
Comment (macaptain) [reply vm-setup/roles/v1aX_integration_test/templates/cluster-template-controlplane-kubeadm-config-centos.yaml:16]: All these commands are happening on the target node. Once the target cluster is provisioned and kubeadm is done, you can `ssh metal3@192.168.111.100` and run `kubectl` commands to query the target Kubernetes cluster on the target node.

Comment (lentzi90) [reply vm-setup/roles/v1aX_integration_test/templates/cluster-template-controlplane-kubeadm-config-centos.yaml:16]: Right, so it's more of a convenience than anything else?

Comment (macaptain) [reply vm-setup/roles/v1aX_integration_test/templates/cluster-template-controlplane-kubeadm-config-centos.yaml:16]: Yes, I don't think there's anything in CI that depends on `kubectl` commands run on the target node as the metal3 user. It's for when you SSH into the node and want to see what the pods are doing.

File: vm-setup/roles/v1aX_integration_test/templates/clusterctl_env_template_centos.rc (hunk 2 above)
Comment (Xenwar) [reply vm-setup/roles/v1aX_integration_test/templates/clusterctl_env_template_centos.rc:29]: So, the vars are moved to a new file 
And, the commands are distributed to relevant files ?

Comment (macaptain) [reply vm-setup/roles/v1aX_integration_test/templates/clusterctl_env_template_centos.rc:29]: Previously the vars came from both this file and the Ansible vars/main.yml file. Now all vars that come from main.yml.

For these commands, these are now included as .yml files into the main templates. Which set of commands is included is dependent on OS and controlplane/worker.

**Review by fmuyassarov — COMMENTED** [review]
I like that way we will manage the environment vars after this change and a single path for all the templates.
/approve

**Pull request merged**